### ISSUE TITLE
Add support for Scala 3

### ISSUE THREAD

**Issue by Flowdalic** [joan38/mill-scalafix#46]
It appears mill-scalafix 0.2.2 does not yet support Scala 3. At least, as soon as I change the scalaVersion to 3, I get

```
Resolution failed for 1 modules:
--------------------------------------------
  org.scalameta:semanticdb-scalac_3.0.0:4.4.10
```

when running `mill` on my project. Commenting out mill-scalafix makes the issue disappear.

**Comment by joan38** [comment]
Thanks for the ticket but for now not even scalafix supports Scala 3 https://github.com/scalacenter/scalafix/issues/1316 and I don't think semanticdb supports Scala 3 yet.

Scala Steward will automatically update these once they are published.

**Comment by olafurpg** [comment]
The semanticdb-scalac compiler plugin does not exist for Scala 3 because the Scala 3 compiler has built-in SemanticDB support. You can remove the dependency on semanticdb-scalac and add `-Xsemanticdb` compiler option instead. The latest release of Scalafix supports Scala 3 https://contributors.scala-lang.org/t/scalafix-0-9-28-with-support-for-scala-3/5091 However, note that Scalafix rules that depend on compiler diagnostics (for example, `RemoveUnused`) don't work with Scala 3 because the compiler doesn't report warnings about unused code.

**Comment by joan38** [comment]
Ah I see, thanks @olafurpg for the info.
I just created this PR: https://github.com/joan38/mill-scalafix/pull/47

**Comment by joan38** [comment]
Released in `0.2.3`
